file_handler.cpp

```diff
@@ -131,11 +130,34 @@ RequestHandler::Status file_handler::HandleRequest(const Request &request, Respo
 	else if (extension == "png") {
 		extension = "image/png";
 	}
+	else if (extension == "md") {
+		extension = "text/markdown";
+	}
 	else {
 		extension = "text/plain";
 	}
 
-	response->AddHeader("Content-Type", extension);
+	if (extension == "text/markdown") {
+		markdown::Document doc;
+		doc.read(body);
+		std::ostringstream stream;
+		doc.write(stream);
+
+		std::string markd = stream.str();
+
+		response->SetBody(markd);
+
+		response->AddHeader("Content-Length", std::to_string(markd.size()));
+
+		response->AddHeader("Content-Type", "text/html");
+
+	} else {
+		response->SetBody(body);
+
+		response->AddHeader("Content-Length", std::to_string(body.size()));
+
+		response->AddHeader("Content-Type", extension);
```

Comment:
Wouldn't these headers already be present in the response? would this overwrite values or would it duplicate headers? 
If it would duplicate headers, we've got to delete the old headers that have been set on this response before this block of code